### DISCUSSION TITLE
Dropdown Triggers Actions Twice Since 4.1.0

### DISCUSSION THREAD

**Discussion by imageaid** [excid3/tailwindcss-stimulus-components#228] [Q&A] [answered]
I ran into an odd issue after updating our app to 5.x from 4.0.4 today and wanted to report it just in case it's legit/good to know, etc. as I did not see anything about this change in the docs (sincerest apologies if I totally overlooked the change being noted in the docs).

Previously, we had the following code for drop-downs, which always worked well: 

```
<div class="relative" data-controller="dropdown">
   <div class="inline-block select-none" aria-label="Profile Menu" data-action="click->dropdown#show click@window->dropdown#hide" role="button">...</div>
</div>

```
After upgrading to 5.x, our drop-downs would open but immediately close. Inspecting the HTML markup I noticed that the newly added call in the `connect()` method of `dropdown.js` to `this.#initializeDropdownActions()` was injecting code into our html so that the resulting HTML markup looks like this: 

```
<div class="relative" data-controller="dropdown" data-action="click->dropdown#toggle click@window->dropdown#hide keydown.up->dropdown#previousItem keydown.down->dropdown#nextItem keydown.esc->dropdown#hide">
   <div class="inline-block select-none" aria-label="Profile Menu" data-action="click->dropdown#show click@window->dropdown#hide" role="button">...</div>
</div>
```

Basically, the code injection appears to not respect the following comment, noted above the `#initializeDropdownActions()` method: "// Note: If existing actions are already specified by the user, they will be preserved and augmented without any redundancy."

The result is that the `dropdown#toggle` method is called twice resulting in the dropdown opening and then immediately closing (i.e., toggle called twice). 

When I remove the `data-action` attribute on the child element (the one our team had added) all works swimmingly. 

Note: I also tried manually adding our `data-action` attribute to the same element as the one with the `data-controller` attribute but that did not resolve the issue.

**Comment by excid3** [comment, marked as answer]
Correct, you have `click -> show` but the initialize adds `click->dropdown` so it would not be able to deduplicate those.

You basically don't need to add any data-action in v5 because it can take care of it for you.

**Comment by imageaid** [comment]
Sounds good. Thanks, as always, Chris! Appreciate all the work!!
